src/coreComponents/physicsSolvers/multiphysics/CoupledSolver.hpp

```diff
@@ -368,13 +353,21 @@ class CoupledSolver : public SolverBase
 
     real64 dtReturnTemporary;
 
+    Timestamp const meshModificationTimestamp = getMeshModificationTimestamp( domain );
+
     forEachArgInTuple( m_solvers, [&]( auto & solver, auto )
     {
-      solver->setupSystem( domain,
-                           solver->getDofManager(),
-                           solver->getLocalMatrix(),
-                           solver->getSystemRhs(),
-                           solver->getSystemSolution() );
+
+      // Only build the sparsity pattern if the mesh has changed
+      if( meshModificationTimestamp > solver->getSystemSetupTimestamp() )
+      {
+        solver->setupSystem( domain,
+                             solver->getDofManager(),
+                             solver->getLocalMatrix(),
+                             solver->getSystemRhs(),
+                             solver->getSystemSolution() );
+        solver->setSystemSetupTimestamp( meshModificationTimestamp );
+      }
```

Comment:
Not sure if it would be better to roll the timestamp check/update into `setupSystem()`. Right now this piece is repeated in every solver that implements a custom solver step function. If moved, it will repeat in every solver that reimplements `setupSystem()` (not sure if there are more or less of these).

The latter can be improved by inverting the flow - always calling `SolverBase::setupSystem()` (i.e. not overriding it), and having it call some `virtual void fillSparsity()` that will be overridden by derived classes to invoke custom sparsity fill kernels. In that case, the timestamp check/bypass logic will only appear in code once. But this refactoring is beyond the scope of this PR.

Reply:
Ok sounds good, I like the idea of having just one `setupSystem` and one place where we check the timestamp. I agree with the plan of doing that in a separate PR.